app/main.py

```diff
@@ -11,7 +11,12 @@ def copy_file(command: str) -> None:
     if file_name_in == file_name_out or cmd != "cp":
         return
 
-    with (open(file_name_in, "r") as file_in,
+    try:
+        open(file_name_in)
+    except (FileExistsError, PermissionError) as e:
```

Comment:
FileExistsError:
FileExistsError is raised when you try to create or rename a file or directory, but a file or directory with the same name already exists in the target location. It indicates that the operation failed because the file or directory already exists. This error is specific to situations where the file or directory you're trying to create or rename already exists.

For example, if you attempt to create a file with the name "example.txt" in a directory where another file with the same name already exists, a FileExistsError will be raised.

FileNotFoundError:
FileNotFoundError is raised when you try to access or perform operations on a file or directory that does not exist in the specified location. It indicates that the file or directory could not be found at the given path.

For example, if you try to open a file for reading with the path "nonexistent.txt," and there is no file with that name in the specified location, a FileNotFoundError will be raised.

Reply:
Nice explanation!